From 12db6b2d83fe9deec1607813a63ee92e135a93c9 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sat, 18 Nov 2023 21:14:56 +0800
Subject: [PATCH] scripts/gcc: properly handle duplicated multilib entries

---
 scripts/build/cc/gcc.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/build/cc/gcc.sh b/scripts/build/cc/gcc.sh
index 275e239c..86388528 100644
--- a/scripts/build/cc/gcc.sh
+++ b/scripts/build/cc/gcc.sh
@@ -850,6 +850,9 @@ gcc_movelibs()
                 ;;
         esac
         if [ -f "${gcc_dir}/${f}" ]; then
+            if [ "${gcc_dir}/${f}" -ef "${dst_dir}/${f}" ]; then
+                continue
+            fi
             CT_DoExecLog ALL mkdir -p "${dst_dir}"
             CT_DoExecLog ALL mv "${gcc_dir}/${f}" "${dst_dir}/${f}"
             CT_DoExecLog ALL ln -sf "${rel}/${dst_dir#${canon_prefix}/}/${f}" "${gcc_dir}/${f}"
-- 
2.42.1

